nekton/nii2dcm.py

```diff
@@ -0,0 +1,157 @@
+            DICOMs based on the instance number.
+
+        Args:
+            dcmfiles (List[Path]): list of path to original dicom files
+            seg (np.ndarray): 3d numpy array of a segmentation
+
+        Returns:
+            List[Path]: sorted list of dicoms based on the order
+        """
+        assert seg.shape[-1] == len(
+            dcmfiles
+        ), f"""Need 1 DICOM per slice of NifTi;
+        Found {len(dcmfiles)} DICOMS for {len(seg)} NifTi slice"""
+
+        z_locs = [pydicom.dcmread(path).InstanceNumber for path in dcmfiles]
+
+        return self.sort_order(z_locs, dcmfiles)
+
+    @staticmethod
+    def _create_dicomseg(
+        seg_map: Dataset, segImage: np.ndarray, dcmImage: FileDataset
+    ) -> SegmentationDataset:
+        """create a dicomseg for storage
+
+        Args:
+            seg_map (Dataset): Dataset info extraced from the mapping json
+            segImage (np.ndarray): a single slice of the segmentation nifti image
+            dcmImage (FileDataset): input dicom to which the dicomseg is to be linked to
+
+        Returns:
+            SegmentationDataset: created dicomseg file
+        """
+        # add fake storage info if necessary
+        appendedImagePosition = False
+        try:
+            dcmImage.ImagePositionPatient
+        except Exception:
+            appendedImagePosition = True
+            dcmImage.ImagePositionPatient = [0, 0, 0]
+
+        # convert to itk image
+        segImage_itk = sitk.GetImageFromArray(segImage.astype(np.uint8))
+
+        writer = pydicom_seg.MultiClassWriter(
+            template=seg_map,
+            inplane_cropping=False,  # Crop image slices to the minimum bounding box on
+            # x and y axes. Maybe not supported by other frameworks.
+            skip_empty_slices=True,  # Don't encode slices with only zeros
+            skip_missing_segment=True,  # If a segment definition is missing in the
+            # template, then raise an error instead of
+            # skipping it.
+        )
+
+        # write the image
+        dcmseg = writer.write(segImage_itk, source_images=[dcmImage])
+        # correct the acquition time and other info if neccesary
+        dcmseg.AcquisitionTime = dcmImage.AcquisitionTime
+        if appendedImagePosition:
+            dcmseg.ImagePositionPatient = None
+
+        return dcmseg
+
+    def multilabel_converter(
+        self, segfiles=List[Path], dcmfiles=List[Path]
+    ) -> List[Path]:
+        # all_individual_segs = [nib.load(file).get_fdata() for file in segfiles]
+        # de-priorisited for now
+
+        raise NotImplementedError(
+            "Multilabel NifTi s to DICOMSeg cannot to be handled yet!!"
+        )
+
+    def multiclass_converter(
+        self, segfile: Path, segMapping: Path, dcmfiles: List[Path]
+    ) -> List[Path]:
+        """Convert a given nifti segmentation to dicomseg for multiclass labels
```

Comment:
terms are a bit confusing, maybe we can use `segmentation` instead of `label` when talking about multiclass in order to not to confuse with multilabel case

Reply:
made all the references to labels as segmentation.